### ISSUE TITLE
Saving models at incremental steps

### ISSUE THREAD

**Issue by ghost** [ContinualAI/avalanche#1016]
🐛 **Describe the bug**
Avalanche is amazing to train models 'from scratch' with multiple incremental learning experiences. For example, imagine MNIST separated with 5 experiences, the training plugins allow to run from the first step (2 classes) up to the final one (10 classes). In order to save some time, I was trying to save the model after each training experiences so if I need to modify the scenario configuration (for example, LwF parametres) I don't need to restart the training from the begging because I previously got the model with the optimal configuration. 

 

🐜 **To Reproduce**
First of all, I was trying to pass the pretrained model (for example, in the first two classes) and start the training in the second experience (next two classes) but this dind't work because LwF doesn't know that the model has been previously trained, so catastrophic forgetting happened. 

My idea was to modify the LwF plugging so for each training experience I can pass the model of the previous experience and the previously trained classes. This works in some way because the distillation loss is calculated but the behavior of the training is still different compared to the normal one (starting from the begging). 

This is how the code looks like. 

```
class LwFPlugin(SupervisedPlugin):
    def __init__(self, alpha, temperature, model, prev_classes):
        super().__init__()
        self.alpha = alpha
        self.temperature = temperature
        self.prev_model = model
        self.prev_classes = prev_classes
        
        #self.prev_model = None
        #self.prev_classes = {"0": set()}
```

🐝 **Expected behavior**
I expect the same behavior loading the pretrained model and starting the training from the first experience.  

🐞 **Screenshots**
When I get some free time, I'm going to reproduce it again so you can see the difference performance in each case. Hope I have explained myself in a clear way.

**Comment by AntonioCarta** [comment]
To support this usecase we would need to support serialization, not just of the entire model but of the strategy. Something like:
```
for e in stream:
    strat.train(e)
    strat.save(fname)
```
reload:
```
strat = strat.load(fname)
```
like in pytorch.

This probably requires some care when saving things like pytorch models or loggers (which may have open files).

**Comment by AntonioCarta** [comment]
Closing this since we now support serialization.
